Remove `ModifiedTxStatus` from the spec and the code (#8210)
tendermint/tendermint: spec/abci++/abci++_tmint_expected_behavior_002_draft.md

As for the new methods:

> * `PrepareProposal` should create a list of [TxRecord](./abci%2B%2B_methods_002_draft.md#txrecord) each containing a

Author: Those are URL encoded all over. I've changed this one following your suggestion. If it works, I'll do another pass and change the rest

  transaction passed in `RequestPrepareProposal.txs`, in the same other. The field `action` must be set to `UNMODIFIED`
  for all [TxRecord](./abci%2B%2B_methods_002_draft.md#txrecord) elements in the list.

>   The Application should check whether the size of all transactions exceeds the byte limit
>   (`RequestPrepareProposal.max_tx_bytes`). If so, the Application should remove transactions at the end of the list
>   until the total byte size conforms to the limit.

Reviewer: ```suggestion
  The Application must check whether the size of all transactions exceeds the byte limit
  (`RequestPrepareProposal.max_tx_bytes`). If so, the Application must remove transactions at the end of the list
  until the total byte size is at or below the limit.
```

As for the new methods:

* `PrepareProposal` must create a list of [TxRecord](./abci++_methods_002_draft.md#txrecord) each containing a
  transaction passed in `RequestPrepareProposal.txs`, in the same other. The field `action` must be set to `UNMODIFIED`
  for all [TxRecord](./abci++_methods_002_draft.md#txrecord) elements in the list.
  The Application must check whether the size of all transactions exceeds the byte limit
  (`RequestPrepareProposal.max_tx_bytes`). If so, the Application must remove transactions at the end of the list
  until the total byte size is at or below the limit.
* `ProcessProposal` must set `ResponseProcessProposal.accept` to _true_ and return.
* `ExtendVote` is to set `ResponseExtendVote.extension` to an empty byte array and return.
* `VerifyVoteExtension` must set `ResponseVerifyVoteExtension.accept` to _true_ if the extension is an empty byte array
  and _false_ otherwise, then return.
* `FinalizeBlock` is to coalesce the implementation of methods `BeginBlock`, `DeliverTx`, `EndBlock`, and `Commit`.

>   The logic extracted from `DeliverTx` should be wrappped by a loop that must execute one iteration per

Reviewer: ```suggestion
  Legacy applications looking to reuse old code that implemented `DeliverTx` should be wrap the legacy
  `DeliverTx` logic in a loop that executes one iteration per
```
Something like this perhaps?
